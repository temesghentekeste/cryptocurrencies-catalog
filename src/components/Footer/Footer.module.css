.mainFooter {
  margin-top: auto;
  padding: 2rem;
  background: #615f5f;
  text-align: center;
  color: #fff;
}

.mainFooter a {
  transition: ease-in-out 0.3s;
}

.mainFooter a:hover {
  color: #f79427;
}

@media screen and (max-width: 500px) {
  .mainFooter {
    padding: 1.5rem 0.4rem;
    font-size: 0.85rem;
  }
}
