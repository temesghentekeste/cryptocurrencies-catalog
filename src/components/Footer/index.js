/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
/* eslint-disable no-undef */
import styles from './Footer.module.css';

const Footer = () => (
  <footer className={styles.mainFooter}>
    <p>&copy; 2021, All Rights Reserved</p>
  </footer>
);

export default Footer;
