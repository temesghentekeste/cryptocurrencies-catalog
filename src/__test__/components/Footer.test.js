import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from '../../components/Footer';

let getByTestId;

beforeEach(() => {
  const component = render(<Footer />);
  getByTestId = component.getByTestId;
});

it('renders the Footer component', () => {
  const component = renderer.create(<Footer />).toJSON();

  expect(component).toMatchSnapshot();
});

test('should render Footer component with correct text', () => {
  const component = renderer.create(<Footer />).toJSON();

  const footerEl = getByTestId('footer');

  expect(footerEl.textContent).toBe('Â© 2021, All Rights Reserved');
});
