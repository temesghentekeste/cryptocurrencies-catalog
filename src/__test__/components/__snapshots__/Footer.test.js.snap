// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Footer component 1`] = `
<footer
  className="mainFooter"
  data-testid="footer"
>
  <p>
    <a
      href="https://www.temesghen.me"
      rel="noreferrer"
      target="_blank"
    >
      ©Temesghen Tekeste  
    </a>
    2021, All Rights Reserved
  </p>
</footer>
`;
